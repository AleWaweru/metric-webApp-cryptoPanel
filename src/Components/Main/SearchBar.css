.search-bar {
  padding: 2rem   1rem;
  float: right;
  right: 10rem;
  display: flex;
  align-items: center;
  background-color: rgb(235, 232, 232);
  height: 40px;
  margin-top: 2rem;
  margin-bottom: 2rem;
  margin-right: 1rem;
  left: 0;
  width: 30%;
  border-radius: 10px;
}

.search-bar input {
  height: 40px;
  padding-left: 7px;
  margin-right: 5px;
  border: none;
  width: 70%;
  background: rgb(235, 232, 232);
  font-size: 18px;
  outline: none;
}

.search-bar button {
  height: 50px;
  border-radius: 10px;
  width: 40%;
  right: 0;
  margin-left: 40%;
  font-size: 20px;
  border: none;
  background-color: rgb(12, 4, 85);
  color: rgb(252, 249, 249);
  cursor: pointer;
}

@media screen and (max-width: 768px) {
  .search-bar {
    width: 90%;
  }

  .search-bar button {
    width: 60%;
  }
}

@media screen and (max-width: 1024px) {
  .search-bar {
    width: 90%;
  }
}
