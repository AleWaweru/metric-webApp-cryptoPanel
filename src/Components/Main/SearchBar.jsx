import React from 'react';
import PropTypes from 'prop-types';
import './SearchBar.css';

const SearchBar = ({
  value, onChange, onKeyDown, onSearchButtonClick,
}) => {
  const handleButtonClick = () => {
    onSearchButtonClick(value);
  };

  return (
    <div className="search-bar">
      <input
        type="text"
        value={value}
        placeholder="Search"
        onChange={onChange}
        onKeyDown={onKeyDown}
      />
      <button type="button" onClick={handleButtonClick}>
        Search
      </button>
    </div>
  );
};

SearchBar.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onKeyDown: PropTypes.func.isRequired,
  onSearchButtonClick: PropTypes.func.isRequired,
};

export default SearchBar;
