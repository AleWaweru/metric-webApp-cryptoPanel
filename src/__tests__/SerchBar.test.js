import React from 'react';
import { render } from '@testing-library/react';
import SearchBar from '../Components/Main/SearchBar';

describe('SearchBar', () => {
  const value = 'Bitcoin';
  const onChange = jest.fn();
  const onKeyDown = jest.fn();
  const onSearchButtonClick = jest.fn();

  it('should render the component correctly', () => {
    const { container } = render(
      <SearchBar
        value={value}
        onChange={onChange}
        onKeyDown={onKeyDown}
        onSearchButtonClick={onSearchButtonClick}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
