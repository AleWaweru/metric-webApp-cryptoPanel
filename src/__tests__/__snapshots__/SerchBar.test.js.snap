// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render the component correctly 1`] = `
<div>
  <div
    class="search-bar"
  >
    <input
      placeholder="Search"
      type="text"
      value="Bitcoin"
    />
    <button
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
